modin/core/io/file_dispatcher.py

```diff
@@ -152,27 +155,15 @@ def read(cls, *args, **kwargs):
         postprocessing work on the resulting query_compiler object.
         """
         query_compiler = cls._read(*args, **kwargs)
-        # TODO (devin-petersohn): Make this section more general for non-pandas kernel
-        # implementations.
-        if StorageFormat.get() == "Pandas":
-            import pandas as kernel_lib
-        elif StorageFormat.get() == "Cudf":
-            import cudf as kernel_lib
-        else:
-            raise NotImplementedError("FIXME")
-
-        if hasattr(query_compiler, "dtypes") and any(
-            isinstance(t, kernel_lib.CategoricalDtype) for t in query_compiler.dtypes
-        ):
-            dtypes = query_compiler.dtypes
-            return query_compiler.astype(
-                {
-                    t: dtypes[t]
-                    for t in dtypes.index
-                    if isinstance(dtypes[t], kernel_lib.CategoricalDtype)
-                },
-                kwargs.get("errors", "raise"),
-            )
+        # TextFileReader can also be returned from `_read`.
```

Comment:
When does this happen?

Reply:
In `test_read_csv_iteration` test, for example.